mono/mini/mini-posix.c

```diff
@@ -210,6 +211,20 @@ MONO_SIG_HANDLER_FUNC (static, sigabrt_signal_handler)
 	}
 }
 
+MONO_SIG_HANDLER_FUNC (static, sigterm_signal_handler)
+{
+	MONO_SIG_HANDLER_INFO_TYPE *info = MONO_SIG_HANDLER_GET_INFO ();
+	MONO_SIG_HANDLER_GET_CONTEXT;
+
+#ifdef HOST_DARWIN
+	if (mono_merp_enabled ())
+		mono_handle_native_crash ("SIGTERM", ctx, info);
+#endif
+
+	mono_chain_signal (MONO_SIG_HANDLER_PARAMS);
+	exit (1);
```

Comment:
Why keeping the `exit (1);`?

Reply:
Without an exit(1), you won't exit? 
I think there's a minor API confusion here. mono_chain_signal only calls another signal handler if it's already been registered. There's no other sigterm handler, so if you put a print between the chain signal and the exit call, it fires. 

```Before chain signal``` And
``` After chain signal, before exit```

Both fire.

This corresponds to the expected state in add_signal_handler, when ```previous_sa.sa_flags & SA_SIGINFO``` is false, and we have no already-registered signal handler to save. 

For this to work the way you want it to, where removing `exit` doesn't make it not `exit`, you'd need to explicitly register the handler at the start of every run that makes mono `exit` when sent sigaction. POSIX just defaults to externally killing a process when there's no signal handler, that doesn't mean that there's automatically a saved signal handler.